### PR TITLE
Filecache unable to decode HPC-5320

### PR DESCRIPTION
**Pull request by itkovian**
- catch the error
- add test to check the shelf is indeed empty

### CODE REVIEW

File: test/cache.py [hunk 1]
```diff
@@ -110,3 +114,22 @@ def test_corrupt_gz_cache(self):
...
+        tempdir = tempfile.mkdtemp()
+        # create a tempfilename
+        (handle, filename) = tempfile.mkstemp(dir=tempdir)
+        f = os.fdopen(handle, 'w')
+        g = gzip.GzipFile(mode='w', fileobj=f)
+        g.write('blabla no json gzip stuffz')
+        g.close()
+
+        e = ValueError('unable to find valid JSON')
+        mock_decode.side_effect = e
```
Comment (JensTimmerman) [inline test/cache.py]: why mock this side effect if you're first going trought the trouble of creating a file that will generate this error in the first place?
Can't you just remove the mocking?

File: lib/vsc/utils/cache.py [hunk 2]
```diff
@@ -100,6 +99,12 @@ def __init__(self, filename, retain_old=True, raise_unpickable=False):
                             self.shelf = {}
                     except (OSError, IOError):
                         self.log.raiseException("Could not load pickle data from %s" % (self.filename,))
+                else:
+                    try:
+                        self.shelf = jsonpickle.decode(s)
+                    except ValueError, err:
+                        self.log.error("Cannot decode JSON from %s [%s]", self.filename, err)
+                        self.shelf = {}
```
Comment (JensTimmerman) [inline lib/vsc/utils/cache.py]: you forgot to ` self.log.info("Cache in %s starts with an empty shelf" % (self.filename,))`

File: test/cache.py (hunk 1 above)
Comment (itkovian) [inline test/cache.py]: Because I want to be 100% certain we end up in the code branch I want to test.

Comment (JensTimmerman) [inline test/cache.py]: how are you certain now?
the method would throw an error anyway? so why mock it?
